src/Analyzers.UnitTests/BuildAnalyzerConfiguration_Test.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Experimental.BuildCop;
+using Shouldly;
+using Xunit;
+
+#nullable disable
```

Comment:
In case the null is met and not tested the test will fail, indicating something is wrong, do you think we need to cover this scenario in Tests as well?

Reply:
@JanKrivanek  since the test will fail once the referenced object is null so author of PR will immedietly know that there is a problem. 
Could you please share your view on this?

Reply:
Which test do you have in mind? Can you mark it as accepting nullables?

If you want to bypass the static analysis check, you can allways stuff the null into nonullable variable via the "`null!`" - perfectly ok for the testing purposes.

Reply:
All infra related tests are now fixed, however I didn't touch the tests that were copied from Roslyn implementation.